File: TalentSystemDefaults/TalentSystemDefaults/DataEntities/DESQLParameter.cs
using System;
using System.Data;
namespace TalentSystemDefaults
{
	/// <summary>
	/// Provides the properties to build sql paramater object
	/// </summary>
	[Serializable()]
	public class DESQLParameter
	{
		#region Class Level Fields
		/// <summary>
		/// Paramater Name
		/// </summary>
		private string _paramName;
		/// <summary>
		/// Parameter Value
		/// </summary>
		private object _paramValue;
		/// <summary>
		/// Parameter Type
		/// </summary>
		private SqlDbType _paramType;
		/// <summary>
		/// Parameter Direction
		/// </summary>
		private ParameterDirection _paramDirection;
		#endregion
		#region Properties
		/// <summary>
		/// Gets or sets the name of the parameter
		/// </summary>
		/// <value>The name of the param.</value>
		public string ParamName
		{
			get
			{
				return _paramName;
			}
			set
			{
				_paramName = value;
			}
		}
		/// <summary>
		/// Gets or sets the parameter value
		/// </summary>
		/// <value>The param value.</value>
		public dynamic ParamValue
		{
			get
			{
				return _paramValue;
			}
			set
			{
				_paramValue = value;
			}
		}
		/// <summary>
		/// Gets or sets the type of the parameter (SqlDbType)
		/// </summary>
		/// <value>The type of the param.</value>
		public SqlDbType ParamType
		{
			get
			{
				return _paramType;
			}
			set
			{
				_paramType = value;
			}
		}
		/// <summary>
		/// Gets or sets the parameter direction of type Data.ParameterDirection
		/// </summary>
		/// <value>The param direction.</value>
		public ParameterDirection ParamDirection
		{
			get
			{
				return _paramDirection;
			}
			set
			{
				_paramDirection = value;
			}
		}
		#endregion
	}
}

File: TalentCommon/Functions/DataEntities/DESQLParameter.vb
''' <summary>
''' Provides the properties to build sql paramater object
''' </summary>
<Serializable()> _
Public Class DESQLParameter

#Region "Class Level Fields"
    ''' <summary>
    ''' Paramater Name
    ''' </summary>
    Private _paramName As String
    ''' <summary>
    ''' Parameter Value
    ''' </summary>
    Private _paramValue As Object
    ''' <summary>
    ''' Parameter Type
    ''' </summary>
    Private _paramType As SqlDbType
    ''' <summary>
    ''' Parameter Direction
    ''' </summary>
    Private _paramDirection As ParameterDirection
#End Region

#Region "Properties"

    ''' <summary>
    ''' Gets or sets the name of the parameter
    ''' </summary>
    ''' <value>The name of the param.</value>
    Public Property ParamName() As String
        Get
            Return _paramName
        End Get
        Set(ByVal value As String)
            _paramName = value
        End Set
    End Property
    ''' <summary>
    ''' Gets or sets the parameter value
    ''' </summary>
    ''' <value>The param value.</value>
    Public Property ParamValue() As Object
        Get
            Return _paramValue
        End Get
        Set(ByVal value As Object)
            _paramValue = value
        End Set
    End Property

    ''' <summary>
    ''' Gets or sets the type of the parameter (SqlDbType)  
    ''' </summary>
    ''' <value>The type of the param.</value>
    Public Property ParamType() As SqlDbType
        Get
            Return _paramType
        End Get
        Set(ByVal value As SqlDbType)
            _paramType = value
        End Set
    End Property

    ''' <summary>
    ''' Gets or sets the parameter direction of type Data.ParameterDirection
    ''' </summary>
    ''' <value>The param direction.</value>
    Public Property ParamDirection() As ParameterDirection
        Get
            Return _paramDirection
        End Get
        Set(ByVal value As ParameterDirection)
            _paramDirection = value
        End Set
    End Property
#End Region

End Class
